community/kernel/src/test/java/org/neo4j/kernel/impl/api/index/inmemory/HashBasedIndex.java

```diff
@@ -288,6 +288,12 @@ public PrimitiveLongIterator query( IndexQuery... predicates )
         }
     }
 
+    @Override
+    public boolean hasFullNumberPrecision()
+    {
+        return true;
+    }
```

Comment:
I think you have failing tests because this isn't true. The HashBasedIndex coerces all numbers to doubles, and thus looses precision.